blocks/src/custom-status/block.js

```diff
@@ -11,19 +11,37 @@ let { SelectControl } = wp.components;
 /**
  * Map Custom Statuses as options for SelectControl
  */
-let statuses = window.EditFlowCustomStatuses.map( s => ({ label: s.name, value: s.slug }) );
+let customStatuses = window.EditFlowStatuses.custom_statuses.map( s => ({ label: s.name, value: s.slug }) );
 
-let getStatusLabel = slug => statuses.find( s => s.value === slug ).label;
+// Trash isn't in `default_statuses` but we need to check for it
+let isCoreStatus = slug => window.EditFlowStatuses.default_statuses.find( s => s.slug === slug ) || slug === 'trash';
+let getCustomStatusLabel = slug => customStatuses.find( s => s.value === slug ).label;
 
 // Hack :(
 // @see https://github.com/WordPress/gutenberg/issues/3144
-let sideEffectL10nManipulation = status => {
-  let node = document.querySelector('.editor-post-save-draft');
-  if ( node ) {
-    document.querySelector( '.editor-post-save-draft' ).innerText = `${ __( 'Save' ) }`
-  }
+// Gutenberg will also override the status set in '.editor-post-save-draft' after save, and there isn't yet a way
+// to subscribe to a "post save" message. So instead set a timeout and override the text in '.editor-post-save-draft'
+let schedulePostStatusUpdater = () => {
+  setTimeout(() => {
+    let status = select( 'core/editor' ).getEditedPostAttribute( 'status' );
+    let statusLabel = getCustomStatusLabel(status);
+
+    if ( typeof status === 'undefined' || typeof statusLabel === 'undefined' || isCoreStatus( status ) ) {
+      return;
+    }
+
+
+    let node = document.querySelector('.editor-post-save-draft');
+    if ( node ) {
+      document.querySelector( '.editor-post-save-draft' ).innerText = `${ __( 'Save as' ) } ${statusLabel}`
+    }
+
+    schedulePostStatusUpdater();
+  }, 250);
 }
 
+schedulePostStatusUpdater();
```

Comment:
I'm a bit worried out the unwanted (and unexpected) effects of having an infinite loop to fix the label bug (i.e. where the save button occasionally flips to "Save Draft" after saving).

It would be better to keep the change to have the save button as generic as possible (i.e. "Save") and live with the occasional text flip. And then work with the Gutenberg project to find a better way to fix this.

Reply:
Yea, maybe there's a better way to approach this while waiting for Gutenberg, I'll mess around a bit. Just seems like a point of confusion for consumers